### PR TITLE
7/30: Add PerspectiveManager and other functionality needed for MVP

### PR DESCRIPTION
**Pull request by annemariecabs**
Add PerspectiveManager (which absorbs the previous PerspectiveAnalysis class) and ContentAnalysis class, which now means that a decision is being made and returned by PerspectiveManager (needed for MVP). Update Perspective demo to reflect this. Change JS code in Perspective demo from manual construction of HTML (by building Strings) to using JS DOM code to build elements.

### CODE REVIEW

File: backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        when(mockAttributeScore.getSummaryScore()).thenReturn(mockScore);
+        when(mockScore.getValue()).thenReturn(fakeScore);
+
+        return mockAttributeScore;
+      }
+    });
+
+    return mockAPI;
+  }
+}
```
Comment (UdayKalra) [inline backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:186]: Need a newline here.

File: backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+  /** 
+   * Returns a mock version of the PerspectiveAPI returns the scores that you set in desiredScores
+   * or which returns -1 when AttributeType scores are set.
+   *
+   * @param desiredScores a map of the scores you want the mockAPI to return 
+   *    (if you don't set any, the mock version of this API will always return -1)
+   * @return the mocked version of the PerspectiveAPI
+   */
+  private static PerspectiveAPI setUpMockAPI(Map<AttributeType, Float> desiredScores) {
```
Comment (UdayKalra) [inline backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:151]: There's a JUnit @Before tag used in the tdd section that can be to reduce potential redundancies and automatically handle some setup versus recalling this.

**Comment by UdayKalra** [comment]
Also the merge conflict with the base branch needs to be resolved.

File: backstory/src/main/java/com/google/sps/data/perspective/PerspectiveManager.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+   */
+  Map<AttributeType, Float> getAnalyses();
+
+  /**
+   * Returns the decision on the appropriateness of the text
+   *
+   * @return true, if content considered appropriate; false, otherwise
+   */
+  boolean getDecision();
+}
```
Comment (UdayKalra) [inline backstory/src/main/java/com/google/sps/data/perspective/PerspectiveManager.java:45]: Newline here.

File: backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java (hunk 1 above)
Comment (annemariecabs) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:183]: Done

File: backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java (hunk 2 above)
Comment (annemariecabs) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:148]: Unfortunately, you have to set up a new mock API every time you want custom values, which could be different each test. That's why it's called every time. As such, I think that creating one mock API at the beginning wouldn't be that helpful. @UdayKalra Do you have a different suggestion as to how to reduce redundancy?

Comment (UdayKalra) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:148]: You can establish api and input as class fields and reinstantiate each test versus redeclaring the typing every test.

Comment (UdayKalra) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:148]: Also this " Map<AttributeType, Float> input = new HashMap<AttributeType, Float>();" can go in a @before

File: backstory/src/main/java/com/google/sps/data/perspective/PerspectiveManager.java (hunk 3 above)
Comment (annemariecabs) [reply backstory/src/main/java/com/google/sps/data/perspective/PerspectiveManager.java]: Done

File: backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java (hunk 2 above)
Comment (annemariecabs) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:156]: Done

Comment (annemariecabs) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:156]: Thank you. That's super helpful. I will fix these.

Comment (annemariecabs) [reply backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java:156]: Done

**Pull request closed without merging**